File: frontend/src/types/lab.ts
export interface LabResult {
  id: number;
  test_name: string;
  value: number;
  unit: string;
  reference_min: number;
  reference_max: number;
  observed_at: string;
  source_document: string;
}

export interface LabListResponse {
  patient_id: number;
  items: LabResult[];
  total: number;
}

File: backend/app/schemas/lab.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class LabResultOut(BaseModel):
    id: int
    test_name: str
    value: float
    unit: str
    reference_min: float
    reference_max: float
    observed_at: datetime
    source_document: str

    model_config = ConfigDict(from_attributes=True)


class LabListResponse(BaseModel):
    patient_id: int
    items: list[LabResultOut]
    total: int
